client/my-sites/plans-features-main/index.tsx

```diff
@@ -326,11 +328,27 @@ const PlansFeaturesMain = ( {
 		// in that case and exit. `FreePlanPaidDomainDialog` takes over from there.
 		// It only applies to main onboarding flow and the paid media flow at the moment.
 		// Standardizing it or not is TBD; see Automattic/growth-foundations#63 and pdgrnI-2nV-p2#comment-4110 for relevant discussion.
-		if ( ( 'onboarding' === flowName || 'onboarding-pm' === flowName ) && ! cartItemForPlan ) {
+		if ( ! cartItemForPlan ) {
+			/**
+			 * Delay showing modal until the relevant experiment has loaded
+			 */
+			if ( 'onboarding-pm' === flowName && isPlanUpsellEnabledOnFreeDomain.isLoading ) {
+				return;
+			} else if (
+				'onboarding' === flowName &&
+				( isCustomDomainAllowedOnFreePlan.isLoading || ! isPlanUpsellEnabledOnFreeDomain.isLoading )
+			) {
+				return;
+			}
+
```

Comment:
Following the tweaks we can do in the hook, since `isLoading` is automatically `false` when the eligible condition is false, we don't need to check the flowName here. 

One memo for myself to check: what happen from the user side? Is there a loading placeholder?